### ISSUE TITLE
Sapiovore cancels out Psychopath (and Killer Drive)

### ISSUE THREAD

**Issue by Stadler76** [CleverRaven/Cataclysm-DDA#58764]
### Describe the bug

When I obtain the Sapiovore trait (e. g. from lizard mutation) I lose the Psychopath trait among others, like Killer Drive.

Looking at [this chunk of consumption.cpp](https://github.com/CleverRaven/Cataclysm-DDA/blob/57fd5acf90b6f2930e5d9655b5d3ac8d44643240/src/consumption.cpp#L1183-L1224) especially at the comment in there reading:
```cpp
// Sapiovores don't recognize humans as the same species.
// But let them possibly feel cool about eating sapient stuff - treat like psycho
// However, spiritual sapiovores should still recognize humans as having a soul or special for religious reasons
```
I highly doubt, that this is intended behaviour.

This might be another victim of the mutation rework.
However: IIRC this worked a couple dev builds before, but I have yet to hunt down the change, that broke it.

### Steps to reproduce

1. ~~Start off with the Psychopath trait or~~ Gain Psychopath by mutating through the debug menu.
2. Go for any mutation that 'grants' the Sapiovore trait, e. g. lizard.
3. Mutate until you get the Sapiovore trait by normal means of through the debug menu.

### Expected behavior

You keep the Psychopath trait when gaining the Sapiovore trait.

### Screenshots

![grafik](https://user-images.githubusercontent.com/13157984/175841811-0cbf322c-5b8c-4dda-9fb9-fc5e3681515e.png)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.1766 (21H2)
- Game Version: d4e5c54 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Magiclysm [magiclysm],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills]
]

**Comment by Stadler76** [comment]
Small update: Just found out, that the opposite is also the case: Psychopath blocks Sapiovore, so with Psychopath as a base trait you won't be able to get Sapiovore.

**Comment by hexagonrecursion** [comment]
/confirmed

I injected a whole bunch of mutagenic catalyst and lizard mutagenic primer and eventually this happened:
![Screenshot from 2022-06-27 20-18-33](https://user-images.githubusercontent.com/52621858/175998603-8daa3311-d674-4f66-b2f3-07501aab4495.png)

I then debugged psychopath back in and kept going until:
![Screenshot from 2022-06-27 20-29-37](https://user-images.githubusercontent.com/52621858/176000501-628ea696-9cab-47d0-9726-3405a498d171.png)

I debugged psychopath back in and continued:
![image](https://user-images.githubusercontent.com/52621858/176003154-e5073eab-985a-4df7-a9d3-a980ee9346f4.png)

And again:
![image](https://user-images.githubusercontent.com/52621858/176003730-13769aa0-2b0f-4b6c-8c1b-e02cc1aa9422.png)

At this point trying to add the psychopath mutation back in via the debug menu had no effect.

- OS: Linux
    - OS Version: <unknown>
- Game Version: cb051eb [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

**Comment by Stadler76** [comment]
For Culler, Hunter and Predator it doesn't always happen, at least not for me:
![grafik](https://user-images.githubusercontent.com/13157984/176036212-ca1b6640-52e5-4593-ac20-85f2d300a032.png)
But I remember, that it did happen at least once for me (not sure which one it was exactly).

**Comment by Stadler76** [comment]
As a heads up: I have this change in mind as well:
```cpp
        } else if( psycho ) {
            add_msg_if_player( _( "Meh.  You've eaten worse." ) );
        } else if( sapiovore ) {
            // Omitting the message for sapiovores. They just don't care what meat they eat.
```

**Comment by Stadler76** [comment]
@fairyarmadillo: Originally I had the following in mind as an at least partial fix for issue #49238:
```diff
-        } else if( cannibal && psycho ) {
+        } else if( ( cannibal || sapiovore ) && psycho ) {
             add_msg_if_player( m_good, _( "You feast upon the human flesh." ) );
             add_morale( MORALE_CANNIBAL, 15, 200 );
```
My thought behind it was, that a `sapiovore && psycho` somewhat thinks, that humans are something tasty.

I could even change this to:
```cpp
        } else if( sapiovore || cannibal && psycho ) {
             add_msg_if_player( m_good, _( "You feast upon the human flesh." ) );
             add_morale( MORALE_CANNIBAL, 15, 200 );
```
considering the bonus to intimidation, which reads to me, like: "Do xyz for me or I'll eat you!"
Not as an actual thread, but more, like a mindset towards humans (Note, that Predator is a requirement for Sapiovore).

Alternatively to my above suggestion, given that +15 morale is too much I could also make it like this:
```cpp
        } else if( psycho ) {
            add_msg_if_player( _( "Meh.  You've eaten worse." ) );
        } else if( sapiovore ) {
             add_msg_if_player( _( "You devour the human meat." ) );
             add_morale( MORALE_CANNIBAL, 5, 50 );
```

I personally prefer the above. Mainly because you won't find that many human flesh to 'feast upon'.

I'll figure something out in a couple hours and make a PR for that.
